client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx

```diff
@@ -0,0 +1,213 @@
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormCheckbox from 'components/forms/form-checkbox';
+import { CALYPSO_CONTACT } from 'lib/url/support';
+import { getName, isRefundable, maybeWithinRefundPeriod } from 'lib/purchases';
+
+class RemoveDomainDialog extends Component {
+	static propTypes = {
+		isRemoving: PropTypes.bool.isRequired,
+		isDialogVisible: PropTypes.bool.isRequired,
+		removePurchase: PropTypes.func.isRequired,
+		closeDialog: PropTypes.func.isRequired,
+		purchase: PropTypes.object,
+	};
+
+	state = {
+		step: 1,
+		domainValidated: false,
+		userAgreed: false,
+		showErrors: false,
+	};
+
+	renderFirstStep( productName ) {
+		const { translate, purchase } = this.props;
+
+		return (
+			<Fragment>
+				<FormSectionHeading>
+					{ translate( '{{strong}}Delete{{/strong}} %(domain)s and remove it from your account.', {
+						args: { domain: productName },
+						components: { strong: <strong /> },
+					} ) }
+				</FormSectionHeading>
+				<p>
+					{ translate(
+						'Deleting {{strong}}%(domain)s{{/strong}} is {{strong}}permanent{{/strong}}. ' +
+							'You will no longer own it, and it could be registered ' +
+							'by someone else.',
+						{
+							args: { domain: productName },
+							components: { strong: <strong /> },
+						}
+					) }
+				</p>
+				{ ! isRefundable( purchase ) && maybeWithinRefundPeriod( purchase ) && (
+					<p>
+						<strong>
+							{ translate(
+								"We're not able to refund this purchase automatically. " +
+									"If you're canceling within %(refundPeriodInDays)s days of " +
+									'purchase, {{contactLink}}contact us{{/contactLink}} to ' +
+									'request a refund.',
+								{
+									args: {
+										refundPeriodInDays: purchase.refundPeriodInDays,
+									},
+									components: {
+										contactLink: <a href={ CALYPSO_CONTACT } />,
+									},
+								}
+							) }
+						</strong>
+					</p>
+				) }
+			</Fragment>
+		);
+	}
+
+	onDomainChange = event => {
+		const productName = getName( this.props.purchase );
+		this.setState( { domainValidated: event.currentTarget.value === productName } );
+	};
+
+	onAgreeChange = event => {
+		this.setState( { userAgreed: event.target.checked } );
+	};
+
+	renderSecondStep( productName ) {
+		const { translate } = this.props;
+
+		return (
+			<Fragment>
+				<FormSectionHeading>
+					{ translate(
+						'{{strong}}Delete{{/strong}} this domain by typing "%(domain)s" into the field below:',
+						{
+							args: { domain: productName },
+							components: { strong: <strong /> },
+						}
+					) }
+				</FormSectionHeading>
+				<FormFieldset>
+					<FormLabel>
+						{ translate( 'Type your domain name to proceed', { context: 'Domain name' } ) }
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?